### ISSUE TITLE
Would you please add touch screen support in the future?

### ISSUE THREAD

**Issue by useame** [RPCS3/rpcs3#7681]
Since computers running Windows 10 are common to be equipped with touch screen (for example, Microsoft Surface) , maybe a virtual control pad on screen would be helpful.

**Comment by useame** [comment]
Well , PPSSPP supports virtual gamepad on its android version, maybe you can also add virtual gamepad support through touch screen in the Windows 10 version of RPCS3.

**Comment by Adiost** [comment]
I feel like this is out of scope of the emulator. It supports xinput, which could be used for any variety of controllers, including a third party on-screen gamepad.

**Comment by dio-gh** [comment]
> computers running Windows 10 are common to be equipped with touch screen

I'd say `common` is a significant overstatement. Most Windows devices are bog standard PCs and laptops, and only a small segment of that is touch-capable - and even from those touch-capable devices, only a fraction is practically usable for gaming purposes. Such a device either has to be a tablet, a convertible, or a Surface product. The former two is by-and-large incapable of running RPCS3 properly, and the latter has a much smaller userbase than you may expect. The only way some could enjoy RPCS3 on the former two is via a streaming application, such as Parsec, but that provides it's own virtual gamepad.

All in all, I agree with Adiost, such a feature is simply way out-of-scope for now and even in the medium-term future. The benefitting userbase is incredibly tiny, and the alternative solutions provide a good enough experience.

**Comment by bevanweiss** [comment]
@useame 
Isn't this 'virtual gamepad' already available via the onscreen keyboard? ;)
If this doesn't get passed through to the keyboard handler then let us know of course.
I think the effort to add in a gamepad popup and have it reliably ONLY work for touch screen controllers (because if it popped up whilst I was gaming with my controller I'd be furious) is unlikely (at this stage) to convince someone to implement this.
It might hit a super long term implementation AFTER ARM support for mobile devices is done... but... the current drift is away from ARM support... so you might need to get involved in the code if you want to see this feature in your lifetime :)

**Comment by ghost** [comment]
Wouldn't this be impossible to pull off? Emulators such as PPSSPP and Vita3K work with touchscreen, because those devices they are emulating are touchscreen devices. However, the PS3 never had anything like this as far as I know. And also what function would this contribute to? This FR just seems like a waste of time if it was ever picked up and worked on IMO.

**Comment by talynone** [comment]
Agreed this is out of scope, and can be achieved with third party apps such as:

https://github.com/Gtxplosiom/TBVGPE
